File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP5_projet_1_Sofiane/rendu/Projet_1_MEDDOURI_VANSCOOR/src/Modele.java
public class Modele{

    //attributs
    
    private String modele;
    
    //constructeur

    public Modele(String modele){
        this.modele=modele;
    }
    //méthodes
    public String getModele(){
        return modele;
    }
    public void setModele(String modele){
        this.modele=modele;
    }

    public String toString(){
        return "nom : " + getModele();
    }
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2/Voiture/Modele.php
<?php
class Modele implements JsonSerializable{
    protected  $modele;

    public function __construct(String $modele = "boiteu"){
        $this->modele = $modele;
    }

    public function getNom(): String{return $this->modele;}

    public function setModele(String $modele = "boiteu"){$this->modele = $modele;}

//Méthodes
    public function __toString() : String {
        return " nom : " . $this->getNom();
    }

    public function jsonSerialize() : array {
        return ['nom' => $this->getNom()];
    }
}
?>